iclib/bno055.py

```diff
@@ -0,0 +1,276 @@
+    GRV_DATA_X_LSB = 0x2E
+    GRV_DATA_X_MSB = 0x2F
+    GRV_DATA_Y_LSB = 0x30
+    GRV_DATA_Y_MSB = 0x31
+    GRV_DATA_Z_LSB = 0x32
+    GRV_DATA_Z_MSB = 0x33
+    TEMP = 0x34
+    CALIB_STAT = 0x35
+    UNIT_SEL = 0x3B
+    OPR_MODE = 0x3D
+
+
+@dataclass
+class BNO055:
+
+    ADDRESS = 0x28
+    i2c_adress: str
+    gpio_address: str
+    gpio_line: int
+
+    def __init__(self, i2c_address: str, gpio_address: str,
+                 gpio_line: int) -> None:
+        global _logger
+        global i2c
+        global gpio_out_imu_reset
+        logging.basicConfig(level=logging.INFO)
+        _logger = logging.getLogger(__name__)
+        i2c = I2C(i2c_address)
+        """original testing i2c_address: /dev/i2c-3"""
+        gpio_out_imu_reset = GPIO(gpio_address, gpio_line, "out")
+        """original testing gpio_address: /dev/gpiochip4"""
+
+    def operation_mode_sel(self) -> None:
+        self.write(Register.OPR_MODE, 0x0C)
+        _logger.info("operation mode set")
+        # write this for NDOF mode: 0x0C
+
+    def write(self, register: Register, data: int) -> None:
+        try:
+            msg = I2C.Message([register, data], read=False)
+            i2c.transfer(self.ADDRESS, [msg])
+        except IOError as e:
+            _logger.error(f"I2C write error: {e}")
+            raise
+
+    def read(self, register: Register,
+             length: int) -> list[int]:
+        try:
+            read_msg = I2C.Message(bytearray(length), read=True)
+            i2c.transfer(self.ADDRESS, [read_msg])
+            return list(read_msg.data)
+        except IOError as e:
+            _logger.error(f"I2C read error: {e}")
+            raise
+
+    def close(self) -> None:
+        i2c.close()
+        gpio_out_imu_reset.close()
+
+    def reset(self) -> None:
+        gpio_out_imu_reset.write(False)
+        gpio_out_imu_reset.write(True)
+        _logger.info("reset")
+
+    def set_units(self) -> None:
+        "Writing 0x00 into UNIT_SEL selects following:"
+        "Acceleration: m/s^2"
+        "Angular rate: dps"
+        "Temp: Celcius"
+        self.write(Register.UNIT_SEL, 0x00)
+        _logger.info("units set")
+
+    def verify_config(self) -> None:
+        mode = self.read(Register.OPR_MODE, 1)[0] & 0x0F
+        """lower 4 bits of OPR_MODE_REG represent the mode"""
+        units = self.read(Register.UNIT_SEL, 1)[0]
+        _logger.info(f"Current mode: {mode}, Units config: {units}")
+
+    def check_calibration(self) -> None:
+        calib_status = self.read(Register.CALIB_STAT, 1)[0]
+        sys_calib = (calib_status >> 6) & 0x03
+        gyro_calib = (calib_status >> 4) & 0x03
+        accel_calib = (calib_status >> 2) & 0x03
+        mag_calib = calib_status & 0x03
+        _logger.info(
+            (
+             f'Calibration - Sys: {sys_calib}/3, '
+             f'Gyro: {gyro_calib}/3, '
+             f'Accel: {accel_calib}/3, '
+             f'Mag: {mag_calib}/3'
+            )
+        )
```

Comment:
return these values. could be a `tuple`, but prefer to be a separate dataclass with attributes `sys_calib`, etc.